### PR TITLE
SED-2555: automation package services

### PR DESCRIPTION
**Pull request by iegorov777**

### CODE REVIEW

**Review by david-stephan — CHANGES_REQUESTED** [review]

File: step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java [hunk 1]
```diff
@@ -130,10 +164,28 @@ public PackageUpdateResult createOrUpdateAutomationPackage(boolean allowUpdate,
             throw new AutomationPackageManagerException("Unable to read automation package", e);
         }
 
-        AutomationPackage oldPackage = getAutomationPackageByName(packageContent.getName(), objectPredicate);
+        AutomationPackage oldPackage;
+        if (explicitOldId != null) {
+            oldPackage = getAutomationPackageById(explicitOldId, objectPredicate);
```
[the remark this answers is not in the document]
Comment (iegorov777) [reply step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java:175]: I've moved this check to `getAutomationPackageById()`, where I prepare different messages for 'not found' and 'not accessible'.

File: step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java [hunk 2]
```diff
@@ -130,10 +164,28 @@ public PackageUpdateResult createOrUpdateAutomationPackage(boolean allowUpdate,
...
+            if (oldPackage == null) {
+                throw new AutomationPackageManagerException("Automation package hasn't been found by id: " + explicitOldId);
+            }
+
+            String expectedName = packageContent.getName();
+            String actualName = oldPackage.getAttribute(AbstractOrganizableObject.NAME);
+            if (!Objects.equals(expectedName, actualName)) {
```
[the remark this answers is not in the document]
Comment (iegorov777) [reply step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java]: I've changed this. Now we can rename the package within the update, but we check that another package with this name (new name) already exists.

**Review by david-stephan — CHANGES_REQUESTED** [review]

**Comment by iegorov777** [comment]
Yes, indeed!
I've lost this line while moving code between local branches.
Fixed.

File: step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java [hunk 3]
```diff
@@ -130,10 +170,28 @@ public PackageUpdateResult createOrUpdateAutomationPackage(boolean allowUpdate,
...
-        AutomationPackage oldPackage = getAutomationPackageByName(packageContent.getName(), objectPredicate);
+        AutomationPackage oldPackage;
+        if (explicitOldId != null) {
+            oldPackage = getAutomationPackageById(explicitOldId, objectPredicate);
+
+            String newName = packageContent.getName();
+            String oldName = oldPackage.getAttribute(AbstractOrganizableObject.NAME);
+            if (!Objects.equals(newName, oldName)) {
+                // the package with the same name shouldn't exist
+                AutomationPackage existingPackageWithSameName = getAutomationPackageByName(newName, objectPredicate);
```
[the remark this answers is not in the document]
Comment (iegorov777) [reply step-automation-packages/step-automation-packages-manager/src/main/java/step/automation/packages/AutomationPackageManager.java:181]: @david-stephan 
Yes, indeed!
I've lost this line while moving the code between local branches.
Fixed.

**Pull request merged**